storage/src/main/java/tech/pegasys/teku/storage/Store.java

```diff
@@ -354,15 +450,68 @@ public VoteTracker getVote(UnsignedLong validatorIndex) {
 
     @CheckReturnValue
     public SafeFuture<Void> commit() {
```

Comment:
Yeah - there's a lot of logic here.  Tried to reorganize a little here, but not sure how much better this is: https://github.com/PegaSysEng/teku/pull/2006/commits/6199f6dff80b7815eb2b6f1354e4e010ca89bb03

It's messy because there are all of the cross-references to the current transaction and underlying `Store`.